cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -166,11 +174,23 @@ private void cleanupQueue() {
       // Queue head might have already been removed. So instead of calling poll, we call remove.
       if (launchingQueue.remove(programRunId)) {
         LOG.info("Removing request with runId {} due to expired retention time.", programRunId);
+        emitMetrics(Constants.Metrics.FlowControl.FLOW_CONTROL_LAUNCHING_RUNS, launchingQueue.size());
       }
-
     }
   }
 
+  private int getRunningCount() {
```

Comment:
Is it cheap enough? We now call it each time we add / remove and item from launching runs. If it's not cheap, it's better to have "running" metric that we don't need to update when run is added /removed from launching

Reply:
Thanks. running metrics no longer is emitted when a launch queue changes.